### ISSUE TITLE
Field definition objects are mutated when creating schema

### ISSUE THREAD

**Issue by dettier** [graphql/graphql-js#164]
It is not possible now to reuse field definition objects when creating schemas.
For example, you have some basic fields you want to define for all your types:
```
const basicFields = {
  _id : ...,
  userId : ...,  
  createdOn : ...
};
```
Following code doesn't work when reusing the same field definition objects:
```
const userType = GraphQLObjectType({
  name : 'User',
  fields : {
    ...basicFields,
    // some user fields here
};

const postType = GraphQLObjectType({
  name : 'Post',
  fields : {
    ...basicFields,
    // some post fields here
};

const schema = new GraphQLSchema(...) // basicFields gets mutated here, throws error if you reuse some object in type fields definitions
```

It was not easy to find why it's not working, error message was not very helpful:
```Error: Post._id args must be an object with argument names as keys.```

This error could easily be avoided by cloning basicFields, but still this may cause some frustration and in general mutating arguments is considered bad.

**Comment by Sandreu** [comment]
Hello btw,

It works to me...  But what I understand of your error message is just that you got wrong with args key in your _id field....

**Comment by Sandreu** [comment]
https://github.com/graphql/graphql-js/blob/4ff563ec8bb16f0e3f79a7f0ed84d3b3511a535b/src/type/__tests__/validation.js#L426-L442

**Comment by dettier** [comment]
This behaviour is now fixed in 0.4.3. Closing this issue.
